lib/rule-tester/rule-tester.js

```diff
@@ -1075,8 +1084,11 @@ class RuleTester {
                                         ].join("\n"));
 
                                         assert.strictEqual(codeWithAppliedSuggestion, expectedSuggestion.output, `Expected the applied suggestion fix to match the test suggestion output for suggestion at index: ${index} on error with message: "${message.message}"`);
-                                    }
-                                });
+                                        assert.notStrictEqual(expectedSuggestion.output, item.code, `The output of a suggestion should differ from the original source code for suggestion at index: ${index} on error with message: "${message.message}"`);
+                                    });
+                                } else {
+                                    assert.fail(`Error should have an array of suggestions or a number for the amount of suggestions. Instead received "${util.inspect(message.suggestions)}" on error with message: "${message.message}"`);
```

Comment:
```suggestion
                                    assert.fail("Test error object property 'suggestions' should be an array or a number");
```

I think this branch relates to invalid `suggestions` in the test case, not to the received actual suggestions on the `message` object.